libs/community/langchain_community/tools/azure_ai_services/azure_translator/azureTranslate.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import annotations
+
+import logging
+import os
+from typing import Any, Optional
+
+from azure.ai.translation.text import TextTranslationClient
+from azure.core.credentials import AzureKeyCredential
+from langchain_core.callbacks import CallbackManagerForToolRun
+from langchain_core.tools import BaseTool
+
+# Setup logging
+logging.basicConfig(level=logging.WARNING)
+logger = logging.getLogger(__name__)
+
+
+class AzureTranslateTool(BaseTool):
+    """
+    A tool that interacts with the Azure Translator API using the SDK.
+
+    This tool queries the Azure Translator API to translate text between languages.
+    It requires an API key and endpoint, which can be set up as described in the
+    Azure Translator API documentation:
+    https://learn.microsoft.com/en-us/azure/ai-services/translator/
+    translator-text-apis?tabs=python
+    """
+
+    translate_key: str = ""
+    translate_endpoint: str = ""
+    translate_client: Any = None  #: :meta private:
+
+    name: str = "azure_translator_tool"
+    description: str = (
+        "A wrapper around Azure Translator API. Useful for translating text between "
+        "languages. Input must be text (str). Ensure to install the "
+        "azure-ai-translation-text package."
+    )
+
+    def __init__(
+        self,
+        *,
+        translate_key: Optional[str] = None,
+        translate_endpoint: Optional[str] = None,
+    ) -> None:
+        """
+        Initialize the AzureTranslateTool with the given API key and endpoint.
+        """
+        translate_key = translate_key or os.getenv("AZURE_OPENAI_TRANSLATE_API_KEY")
+        translate_endpoint = translate_endpoint or os.getenv(
+            "AZURE_OPENAI_TRANSLATE_ENDPOINT"
+        )
+
+        if not translate_key or not translate_endpoint:
+            raise ValueError("Missing API key or endpoint for Azure Translator API.")
+
+        super().__init__(
+            translate_key=translate_key, translate_endpoint=translate_endpoint
+        )
+
+        self.translate_client = TextTranslationClient(
+            endpoint=translate_endpoint, credential=AzureKeyCredential(translate_key)
+        )
+
+    def _translate_text(self, text: str, to_language: str) -> str:
```

Comment:
Since this gets called internally, I don't believe the user has the ability to pass `to_language`. I think we can follow the way speech-to-text handles this by setting a default on the class: https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/tools/azure_ai_services/speech_to_text.py#L29